### PR TITLE
[DSLX:stdlib] Add `std::distinct`.

### PR DESCRIPTION
**Pull request by cdleary**

### CODE REVIEW

File: docs_src/dslx_std.md [hunk 1]
```diff
@@ -1188,6 +1188,15 @@ index is used in a match expression (which will eagerly evaluate all of its
...
 ultimately discarded from the unselected match arm.
 
+#### `std::distinct`
+
+```dslx-snippet
+pub fn distinct<COUNT: u32, N: u32, S: bool>(items: xN[S][N][COUNT], valid: bool[COUNT]) -> bool
+```
+
+Returns whether all the `items` are distinct (i.e. there are no duplicate
+items) after the `valid` mask is applied.
```
Comment (ericastor) [inline docs_src/dslx_std.md:1198]: This could possibly use an example or two, to clarify what it means to apply the `valid` mask in this case.

**Review by ericastor — COMMENTED** [review]
Looks good! Will approve with an example or two added to the docs, since it seems worth explaining the mask a bit further.

File: docs_src/dslx_std.md (hunk 1 above)
Comment (cdleary) [reply docs_src/dslx_std.md:1198]: I duplicated a unit test into the doc -- I'm not sure it's generally wise. Ideally we'd be able to link to a searched piece of text in the stdlib but github isn't capable of doing that. Really we need a rustdoc like system for DSLX to make this proposition more sustainable. See also #1049

**Review by ericastor — APPROVED** [review]

**Comment by cdleary** [comment]
Thanks Eric! I'm also going to keep an eye out for rustdoc-style opportunities in case there are two-birds one-stone kind of situations there.

**Pull request merged**